### ISSUE TITLE
RetinaNet class accuracy not improving for specific class

### ISSUE THREAD

**Issue by rohansuresh27** [fizyr/keras-retinanet#1288]
I have been training a RetinaNet object detection model for 3 classes where 2 of the 3 classes learn well and give an mAP accuracy of above 90% on test set but the 3rd class gives around 10-15% and fails to detect. All the class annotations are balanced. Have tried changing the input image size and training.

Important Note : The two high accuracy classes have bounding box annotation of similar size which are smaller as compared to the 3rd class.

Analogical Example : 2 classes have annotation the size of tyre of a car and the 3rd class annotation is the size of the entire car.

What do you think could be the reasons for this happening ?

Are there ways I can improve the accuracy of this 3rd class ?

Would really appreciate any inputs from your intuition. Thanks a lot !!

**Comment by hgaiser** [comment]
You could inspect your data using debug.py to verify that the network is capable of detecting the third class properly (ie. do anchors exist that match close enough to the third class). If no anchors match, or only some of them, then the network will fail to detect these.

Strangely enough this generally happens with very small or very elongated objects, but a car shaped object is typically well detected.

**Comment by rohansuresh27** [comment]
This helped. My object falls under the elongated type, hence the issue. 
Thanks a lot !!

**Comment by sindhura234** [comment]
hi @rohansuresh27 how did you solve this problem?
